starter-code/routes/movies.js

```diff
@@ -0,0 +1,70 @@
+const express = require('express');
+const router  = express.Router();
+const Movie = require('../models/movie');
+
+router.get('/', function(req,res,next){
+  Movie.find()
+  .then(function(movies){
+    res.render('/movies/index', {
+      movies: movies
+    })
+  })
+  .catch(err => next(err));
+})
+
+router.get('/new', function(req,res,next){
+  res.render('movies/new')
+})
+
+router.post('/new', function(req,res,next){
+  const title = req.body.title;
+  const genre = req.body.genre;
+  const plot = req.body.plot;
+  const cast = req.body.cast;
+
+  Movie.create({
+    title: title,
+    genre: genre,
+    plot: plot,
+    cast: cast
+  }).then(
+    res.redirect('/movies')
+  )
+})
+//Iteration 10  render du cast? 
+router.get('/:id', function(req,res,next){
+  Movie.findById(req.params.id)
+  .populate('cast')
+  .then(function(celebrity){
+    res.render('movies/show',{
+      movie:movie
+    })
+
+  })
+  .catch(err => next(err));
+})
+//Iteration 11
+router.post('/:id/delete', function(req,res,next) {
+  Movie.findByIdAndRemove(req.params.id)
+    .then(() => {
+      res.redirect('/movies')
+    }).catch(err => next(err));
+})
+//Iteration 12
```

Comment:
Et sur cette itération, c'est une fusion de find() sur les 2 models : 
router.get("/:id/edit", (req, res, next) => {
  Celebrity.find()
    .then(celebrities => {
      Movie.findById(req.params.id)
        .populate("cast")
        .then(movie => {
       //DO SMTHG

          res.render("movies/edit", {
            movie,
            celebrities
          });
        })
        .catch(err => next(err));
    })
    .catch(err => next(err));
});